Add additional volume parameter (#3156)
kyma-project/kyma-environment-broker: docs/contributor/03-71-dynamic-volume-sizes.md

By default, the node volume size for every plan is a static value defined in the plan configuration. When the Dynamic Volume Sizes feature is enabled, Kyma Environment Broker (KEB) reads the volume size per machine type from the Kyma Consumption Reporter (KCR) ConfigMap instead, so that larger machines automatically receive appropriately sized disks.

> Users can also request extra disk space on top of the default by setting the optional `additionalVolumeSizeGi` parameter. KEB computes the total volume size as `defaultVolumeSize + additionalVolumeSizeGi`.

Reviewer: ```suggestion
Users can also request extra disk space on top of the default by setting the optional **additionalVolumeSizeGi** parameter. KEB computes the total volume size as the sum of **defaultVolumeSize** and **additionalVolumeSizeGi**.
```

The feature is controlled by the following environment variables:

| Variable | Default | Description |
|---|---|---|
| **APP_BROKER_DYNAMIC_VOLUME_SIZE_ENABLED** | `false` | Enables dynamic volume size lookup. When `false`, the static plan default is used. |
| **APP_BROKER_KCR_CONFIG_MAP_NAME** | `consumption-reporter-config` | Name of the ConfigMap in the `kcp-system` namespace that provides the volume sizes. |

> | **APP_BROKER_ADDITIONAL_VOLUME_SIZE_GI_PLANS** | _(empty)_ | Comma-separated list of plan names for which the `additionalVolumeSizeGi` parameter is exposed in provisioning and update schemas. Leave empty to disable the feature. Requires `APP_BROKER_DYNAMIC_VOLUME_SIZE_ENABLED` to be `true`. |

Reviewer: ```suggestion
| **APP_BROKER_ADDITIONAL_VOLUME_SIZE_GI_PLANS** | None | Comma-separated list of plan names for which the **additionalVolumeSizeGi** parameter is exposed in provisioning and update schemas. Leave empty to disable the feature. Requires **APP_BROKER_DYNAMIC_VOLUME_SIZE_ENABLED** to be `true`. |
```

## Behavior

### Provisioning

When a new runtime is provisioned, the volume size for the Kyma worker pool and all additional worker pools is computed as follows:

> 1. The base volume size is read from the KCR ConfigMap for each machine type (when `APP_BROKER_DYNAMIC_VOLUME_SIZE_ENABLED` is `true`), or the static plan default is used.

Reviewer: ```suggestion
1. The base volume size is read from the KCR ConfigMap for each machine type (when **APP_BROKER_DYNAMIC_VOLUME_SIZE_ENABLED** is `true`), or the static plan default is used.
```

> 2. If `additionalVolumeSizeGi` is set in the request, it is added to the base volume size.

Reviewer: ```suggestion
2. If **additionalVolumeSizeGi** is set in the request, it is added to the base volume size.
```

### Update

When an existing runtime is updated, the following actions take place:

> - Kyma worker pool: If the machine type changes or `additionalVolumeSizeGi` changes, the volume is recomputed as `base + additionalVolumeSizeGi`. If neither changes, the existing volume is preserved.

Reviewer: ```suggestion
- Kyma worker pool: If the machine type changes or **additionalVolumeSizeGi** changes, the volume is recomputed as the sum of **defaultVolumeSize** and **additionalVolumeSizeGi**. If neither changes, the existing volume is preserved.
```

> - Additional worker pools: The volume is recomputed for pools where the machine type or `additionalVolumeSizeGi` is new or changed compared to the previous operation. Unchanged pools preserve their existing volume.

Reviewer: ```suggestion
- Additional worker pools: The volume is recomputed for pools where the machine type or **additionalVolumeSizeGi** is new or changed compared to the previous operation. Unchanged pools preserve their existing volume.
```
